psl/builtin-connectors/src/mysql_datamodel_connector.rs

```diff
@@ -160,10 +160,14 @@ impl Connector for MySqlDatamodelConnector {
         64
     }
 
-    fn referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction> {
+    fn referential_actions(&self, _relation_mode: &RelationMode) -> BitFlags<ReferentialAction> {
         use ReferentialAction::*;
 
-        relation_mode.allowed_referential_actions(Restrict | Cascade | SetNull | NoAction | SetDefault)
+        Restrict | Cascade | SetNull | NoAction | SetDefault
+    }
+
+    fn emulated_referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction> {
+        relation_mode.allowed_emulated_referential_actions_default()
     }
```

Comment:
no need to change this in this PR, but the default could go as a method on `Connector` itself directly that is only overriden on postgres and sqlite. That way it doesn't need to be on `RelationMode` itself. It would be a more natural place for that logic.

Reply:
I had originally tried that (à-la C++ with virtual functions), but it seems that overriding default methods on `traits` is not supported by Rust. Is there something else I'm missing?

Reply:
It's definitely supported, what error did you get?

Reply:
Alright, I'll try again, maybe I had touched some other things that mislead me

Reply:
Extra note: if we do that, we can drop the `relation_mode` argument on `emulated_referential_actions()`.

Reply:
Nice catch, done